app/components/works/detail_component.rb

```diff
@@ -64,5 +64,10 @@ def keywords
     def draft?
       work.first_draft?
     end
+
+    sig { returns(T::Boolean) }
+    def display_approval?
+      work.pending_approval? && !helpers.allowed_to?(:review?, work)
```

Comment:
i'm unfamiliar with `helpers.allowed_to?(XXXX)` ... where is that defined/coming from?

Reply:
`helpers` is a container object that is available to all view components, and `allowed_to?` is a helper method set up by ActionPolicy.

Reply:
ok, so `app/policies/work_policy.rb` is being used (because we are asking about a `work`) and we are hitting the `review?` method defined there

i'm familiar with the traditional rails helpers available to views, so there is an equivalent for components but stored elsewhere?  thanks

Reply:
@peetucket :speech_balloon: 
> ok, so `app/policies/work_policy.rb` is being used (because we are asking about a `work`) and we are hitting the `review?` method defined there

Yep!
 
> i'm familiar with the traditional rails helpers available to views, so there is an equivalent for components but stored elsewhere? thanks

Yeah, they're the same helpers, and the ViewComponent library pulls them in via this `helpers` method which is available to all view components: https://github.com/github/view_component/blob/ec91c69a0552b90e5d6a891ebb5dd642ed44f2f8/lib/view_component/base.rb#L117-L121

Reply:
thanks for the explanations and sorry if missed these earlier

and it looks like actionpolicy (and the state machine used for works/collections) are gems and not core rails

Reply:
Correct!